wts.go

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) 2014-2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the license is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gowin32
+
+import (
+	"github.com/winlabs/gowin32/wrappers"
+
+	"fmt"
+	"net"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+// WTSConnectState enum type - Go version of WTS_CONNECTSTATE_CLASS
+type WTSConnectState uint32
+
+const (
+	WTSConnectStateActive       WTSConnectState = wrappers.WTSActive
+	WTSConnectStateConnected    WTSConnectState = wrappers.WTSConnected
+	WTSConnectStateConnectQuery WTSConnectState = wrappers.WTSConnectQuery
+	WTSConnectStateShadow       WTSConnectState = wrappers.WTSShadow
+	WTSConnectStateDisconnected WTSConnectState = wrappers.WTSDisconnected
+	WTSConnectStateIdle         WTSConnectState = wrappers.WTSIdle
+	WTSConnectStateListen       WTSConnectState = wrappers.WTSListen
+	WTSConnectStateReset        WTSConnectState = wrappers.WTSReset
+	WTSConnectStateDown         WTSConnectState = wrappers.WTSDown
+	WTSConnectStateInit         WTSConnectState = wrappers.WTSInit
+)
+
+// WTSClientProtocolType enum type go version of WTSClientProtocolType
+type WTSClientProtocolType uint32
+
+const (
+	WTSClientProtocolConsoleSession WTSClientProtocolType = 0
+	WTSClientProtocolInternal       WTSClientProtocolType = 1
+	WTSClientProtocolRDP            WTSClientProtocolType = 2
+)
+
+// WTSClientInfo - go version of WTSCLIENT structure
+type WTSClientInfo struct {
+	ClientName          string
+	Domain              string
+	UserName            string
+	WorkDirectory       string
+	InitialProgram      string
+	EncryptionLevel     byte
+	ClientAddressFamily AddressFamily
+	ClientAddress       [wrappers.CLIENTADDRESS_LENGTH + 1]uint16
```

Comment:
Should we make a structure for the client address instead of representing it as an array?

Reply:
I have no idea how to change ClientAddress. The content depends on AddressFamily but I do not know format of returned values. I checked only the values for IPv4 -that's why I created func (ci *WTSClientInfo) ClientAddressToIP().

Reply:
How about if we have a structure that holds the array of `uint16` but as a private member?  That way, if we decide that we want to change it to some other representation in the future, we can do so without modifying the publicly exposed interface.

Reply:
Ok. Changed.